spec/spec/features/user_posts_index_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe 'User Post Index Page', type: :feature do
+  let(:user) { User.first }
+  before do
+    visit user_posts_path(user)
+  end
+
+  it 'shows the user and posts details' do
+    expect(page).to have_selector("img[src*='#{user.photo}']")
+    expect(page).to have_content(user.name)
+    expect(page).to have_content("Number of posts: #{user.posts.count}")
+
+    user.posts.each do |post|
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.text.truncate(100)) # Assuming you show a truncated body
+      expect(page).to have_content("Comments: #{post.comments.count}")
+      expect(page).to have_content("Likes: #{post.likes.count}")
+      post.recent_comments.each do |comment|
+        expect(page).to have_content(comment.user.name)
+        expect(page).to have_content(comment.text)
+        unique_comment_content = "#{comment.user.name}: #{comment.text}"
+        expect(page).to have_content(unique_comment_content)
+      end
+    end
+  end
+
+  it 'redirects to a post show page when a post is clicked' do
+    post = user.posts.first
+    click_on post.title
+    expect(current_path).to eq(user_post_path(user, post))
```

Comment:
Nice job so far👍🏼. As a[ requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md), kindly add a Test that asserts you can see a section for pagination if there are more posts than fit on the view👍🏼.

<img width="851" alt="image" src="https://github.com/chudisoft/RubyBlog/assets/64643995/f1c14753-305c-4e5a-bfba-fe67ab865b85">

Reply:
- [ ] As pointed out by the previous reviewer, Kindly add test that check if the pagination button is present when there are more posts to follow the project requirement.